README.md

```diff
@@ -11,8 +11,43 @@ The primary objectives of the e2e tests are to ensure a consistent and reliable
 behavior of the kubernetes code base, and to catch hard-to-test bugs before
 users do, when unit and integration tests are insufficient.
 
+## Usage with Canonical K8s
 
-## Usage
+This charm can be used to run tests on Canonical K8s.
+
+First, deploy k8s-operator and k8s-worker, and relate them.
+
+```
+juju deploy k8s
+juju deploy k8s-worker
+juju integrate k8s k8s-worker:cluster
+```
+
+Next, deploy the charm-kubernetes-e2e charm:
+
+```
+juju deploy kubernetes-e2e
+```
+
+Export the kubeconfig file of your Canonical K8s cluster:
+
+```
+juju run k8s/0 get-kubeconfig > kubeconfig
+```
```

Comment:
this won't produce a valid `kubeconfig` will it?

Reply:
I think this provides a yaml like:

```yaml
kubeconfig : |-
    ... string content of kubeconfig file
```

maybe

```suggestion
juju run k8s/0 get-kubeconfig | yq '.kubeconfig' -r> kubeconfig
```

Reply:
Do you think that should be changed on the k8s-operator side to provide a valid kubeconfig?

Reply:
a juju action can only return `key:value` fields -- where each key/value is a str

https://ops.readthedocs.io/en/latest/#ops.ActionEvent.set_results

I guess i'm saying there's nothing the `k8s-operator` can do any better than this

Reply:
On it !